flang/lib/Lower/SymbolMap.h

```diff
@@ -229,7 +229,13 @@ class SymMap {
   /// Find `symbol` and return its value if it appears in the current mappings.
   SymbolBox lookupSymbol(semantics::SymbolRef sym) {
     auto iter = symbolMap.find(&*sym);
-    return (iter == symbolMap.end()) ? SymbolBox() : iter->second;
+    if (iter != symbolMap.end())
+      return iter->second;
+    // Follow host association
+    if (const auto *details =
+            sym->detailsIf<Fortran::semantics::HostAssocDetails>())
+      return lookupSymbol(details->symbol());
+    return SymbolBox();
```

Comment:
I probably did this originally, but I wonder if something like `{None}` would be clearer?

Reply:
I agree it conveys better that it is empty, changed to `return SymbolBox::None{};` (`{None}` did not work).